src/sidebar/components/PendingUpdatesNotification.tsx

```diff
@@ -29,10 +32,12 @@ function PendingUpdatesNotification({
   const store = useSidebarStore();
   const pendingUpdateCount = store.pendingUpdateCount();
   const hasPendingChanges = store.hasPendingUpdatesOrDeletions();
-  const applyPendingUpdates = useCallback(
-    () => streamer.applyPendingUpdates(),
-    [streamer],
-  );
+  const applyPendingUpdates = useCallback(() => {
+    streamer.applyPendingUpdates();
+    api.analytics.events
+      .create({}, { event: 'APPLY_PENDING_UPDATES' })
+      .catch(() => console.error('Could not track applying updates.'));
```

Comment:
I think it would be worth creating a service to abstract the details of calling the API with analytics events. This can make testing easier and also potentially be used to simplify debugging (eg. by having a way to log all events that are recorded).

Reply:
I was planning to create a new service wrapping `APIService` which internally does the above (more or less). Is that what you are suggesting? Or do you mean to also extract the whole logic to perform the HTTP request and such?

Reply:
> I was planning to create a new service wrapping APIService which internally does the above (more or less). Is that what you are suggesting?

Yes, exactly.